src/icp_prototype_backend/src/lib.rs

```diff
@@ -62,6 +63,33 @@ impl ToU64Hash for AccountIdentifier {
     }
 }
 
+fn get_network() -> Network { 
+    NETWORK.with(|net| {
+        *net.borrow()
+    })
+}
+
+fn authenticate() -> Result<(), String> {
```

Comment:
Consider implementing this: https://github.com/AlphaCQ/IC_Utils/blob/main/src/guard.rs

It's pretty interesting.

Reply:
implemented and tested guards